my_enumerables.rb

```diff
@@ -0,0 +1,190 @@
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each(pro = nil)
+    return to_enum unless block_given?
+
+    arr = to_a
+    i = 0
+
+    if pro
+      while i < arr.length
+        proc.cal arr[i]
+        i += 1
+      end
+    else
+      while i < arr.length
+        yield arr[i]
+        i += 1
+      end
+    end
+    arr
+  end
+
+  def my_each_with_index(pro = nil)
+    return to_enum unless block_given?
+
+    arr = to_a
+    i = 0
+
+    if pro
+      while i < arr.length
+        proc.cal arr[i], i
+        i += 1
+      end
+    else
+      while i < arr.length
+        yield arr[i], i
+        i += 1
+      end
+    end
+
+    arr
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = to_a
+    arr2 = []
+    arr.my_each_with_index do |_item, index|
+      arr2.push(arr[index]) if yield arr[index]
+    end
+    arr2
+  end
+
+  def my_all?(args = nil)
+    return to_enum unless block_given? || !args.nil?
+
+    arr = to_a
+    if block_given?
+      arr.my_each_with_index do |_item, index|
+        return false unless yield arr[index]
+      end
+    elsif args.is_a? Class
+      arr.my_each_with_index do |item, _index|
+        return false unless item.class.ancestors.include?(args)
+      end
+    elsif args.is_a? Regexp
+      arr.my_each_with_index do |item, _index|
+        return false unless item.match(args)
+      end
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    return to_enum unless block_given? || !args.nil?
+
+    arr = to_a
+    any = false
+
+    if block_given?
+      arr.my_each_with_index do |_item, index|
+        any = true if yield arr[index]
+      end
+    elsif args.is_a? Class
+      arr.my_each_with_index do |item, _index|
+        puts item.class.ancestors
+        any = true if item.class.ancestors.include?(args)
+      end
+    elsif args.is_a? Regexp
+      arr.my_each_with_index do |item, _index|
+        any = true if item.match(args)
+      end
+    end
+    any
+  end
+
+  def my_none?(args = nil)
+    return to_enum unless block_given? || !args.nil?
+
+    arr = to_a
+    if block_given?
+      arr.my_each_with_index do |_item, index|
+        return true unless yield arr[index]
+      end
+    elsif args.is_a? Class
+      arr.my_each_with_index do |item, _index|
+        return true unless item.class.ancestors.include?(args)
+      end
+    elsif args.is_a? Regexp
+      arr.my_each_with_index do |item, _index|
+        return true unless item.match(args)
+      end
+    end
+    false
+  end
+
+  def my_count(arg = nil)
+    index = 0
+    count = 0
+
+    if arg
+      while index < length
+        count += 1 if proc.call self[index]
+        index += 1
+      end
+    elsif block_given?
+      while index < length
+        count += 1 if yield self[index]
+        index += 1
+      end
+    else
+      my_each do |_item|
+        count += 1
+        index += 1
+      end
+    end
+    count
+  end
```

Comment:
#my_count counts the number of items in enum that are equal to item if an argument is given